File: java/src/main/java/mx/iteso/adapter/WordDocument.java
package mx.iteso.adapter;

/** Word Document.*/
public class WordDocument implements IWordDocument {

    /** Font.*/
    private Font font;

    /** Background Image.*/
    private BackgroundImage backgroundImage;

    /** License.*/
    private MSLicense msLicense;

    /** MS Office Version.*/
    private float version;

    /** Format.*/
    private Format format;

    /** WordDocument. */
    public WordDocument() {
        this.font = new Font("Calibri");
        this.msLicense = new MSLicense("MS License");
        this.format = new Format(this.font, "Bold");
        this.backgroundImage = new BackgroundImage(new Image("imageUrl"));
    }

    /** @return Format.*/
    public Format getFormat() {
        return this.format;
    }

    /** @return Background.*/
    public Image getBackground() {
        return this.backgroundImage.getImage();
    }

    /** @param msOfficeVersion .*/
    public void setMSOfficeVersion(final float msOfficeVersion) {
        this.version = msOfficeVersion;
    }

    /** @return License.*/
    public MSLicense getLicense() {
        return this.msLicense;
    }

    /** @return Font.*/
    public Font getFont() {
        return this.font;
    }

    /** @return Version.*/
    public float getVersion() {
        return this.version;
    }

    /** @return License
     * @param msLic .*/
    public boolean restrictEditIfLicenseIsInvalid(final MSLicense msLic) {
        return this.msLicense == msLic;
    }
}

File: csharp/Adapter Practice/WordDocument.cs
namespace adapter_practice
{
    public class WordDocument : IWordDocument {

        private readonly Font _font;

        private readonly BackgroundImage _backgroundImage;

        private readonly MsLicense _msLicense;

        private float _version;

        private readonly Format _format;

        public WordDocument() {
            _font = new Font("Arial");
            _msLicense = new MsLicense("MS License");
            _format = new Format(_font, "Bold");
            _backgroundImage = new BackgroundImage(new Image("imageUrl"));
        }

        public Format GetFormat() {
            return _format;
        }

        public Image GetBackground() {
            return _backgroundImage.GetImage();
        }

        public void SetMsOfficeVersion(float msOfficeVersion) {
            _version = msOfficeVersion;
        }

        public MsLicense GetLicense() {
            return _msLicense;
        }

        public Font GetFont() {
            return _font;
        }

        public float GetVersion() {
            return _version;
        }

        public bool RestrictEditIfLicenseIsInvalid(MsLicense msLic) {
            return _msLicense == msLic;
        }
    }

}